app/models/manageiq/providers/hawkular/inventory/parser/middleware_domains.rb

```diff
@@ -40,7 +40,7 @@ def fetch_domain_availabilities
     end
 
     def process_domain_availability(availability = nil)
-      return 'unknown' if availability.nil?
+      return 'Unknown' if availability.nil? || availability.first.nil?
```

Comment:
I think the availability.first.nil? is a case that it won't happen from where this method is invoked.
But it is good to have this validation as is more clear now.